nvtabular/ops/groupby.py

```diff
@@ -184,7 +186,14 @@ def _compute_dtype(self, col_schema, input_schema):
         dtype = col_schema.dtype
         is_list = col_schema.is_list
 
-        dtypes = {"count": numpy.int32, "mean": numpy.float32}
+        dtypes = {
+            "count": numpy.int32,
+            "nunique": numpy.int32,
```

Comment:
@radekosmulski not sure we can do `nunique` agg with groupby. did you test it? thanks.

Reply:
With the changes in this PR `nunique` and a couple of other aggs work now 🙂 

![image](https://user-images.githubusercontent.com/2444926/173339064-acabdc51-5e40-48ef-aa3a-d3da5aa3003f.png)

But please don't merge it just yet -- the tests are passing in CI because the transformations don't run on a GPU, but when you run the tests for `Groupby` on a GPU, they fail.

Problem is, I didn't get a good handle on how to fix this today. The list transformation that is throwing the error upon schema dtype verification is different than these other groupby aggs, 

I don't even know where to start looking for where the issue might lie apart from just trying to figure out what is happening where via doing `set_trace` in various places in code... Will think on how to approach this, maybe asking someone to pair on this or to give me some pointers would be a good idea here 🙂

Reply:
The casting that you mentioned, is what I have done for the other aggregations to make them work, but the `list` one is a bit more tricky.

I can have a list of `object`s
![image](https://user-images.githubusercontent.com/2444926/173490381-012443d8-28ab-4381-bc72-9008eca6f316.png)

And also this seems to fail only when run on the CPU but works on the GPU

![image](https://user-images.githubusercontent.com/2444926/173490534-9ee0d713-60fa-48dd-85d1-c6a2a211a407.png)

There is something much more tricky going on here 🙂 

BTW just going to a list of strings works
![image](https://user-images.githubusercontent.com/2444926/173490715-654f7a45-5aaf-4d70-bda6-cbabd2908236.png)

I think ideally what we might want is for the list op to honor the original dtype of the column, but somewhere, somehow, `float32` still gets cast to `float64` on the CPU (I guess it must be something `dask_cudf` is doing) but I am a bit unsure where to start unraveling this in our code.

Maybe another session of looking at `_apply_aggs` from ops.grouby might help, but doing this hasn't been very efficient for me so far 🙂

Reply:
Yeah, honoring the original dtypes makes sense to me too, but as you said, it's tricky to figure out exactly when and where dtypes are modified by the underlying libraries and code. Casting floats to a consistent precision is kind of a workaround for that.

I can see how casting list columns is trickier, that might require some thought. 🤔